// @talkjs/react/dist/talkjs-react.js
import Ne, { createContext as _r, useContext as Or, useState as V, useEffect as D, useMemo as jr, useRef as $e } from "react";
import ae from "talkjs";
function xr(r) {
  return r && r.__esModule && Object.prototype.hasOwnProperty.call(r, "default") ? r.default : r;
}
var oe = { exports: {} }, U = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Ie;
function Cr() {
  if (Ie)
    return U;
  Ie = 1;
  var r = Ne, n = Symbol.for("react.element"), o = Symbol.for("react.fragment"), i = Object.prototype.hasOwnProperty, u = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, f = { key: !0, ref: !0, __self: !0, __source: !0 };
  function v(b, g, R) {
    var y, O = {}, h = null, k = null;
    R !== void 0 && (h = "" + R), g.key !== void 0 && (h = "" + g.key), g.ref !== void 0 && (k = g.ref);
    for (y in g)
      i.call(g, y) && !f.hasOwnProperty(y) && (O[y] = g[y]);
    if (b && b.defaultProps)
      for (y in g = b.defaultProps, g)
        O[y] === void 0 && (O[y] = g[y]);
    return { $$typeof: n, type: b, key: h, ref: k, props: O, _owner: u.current };
  }
  return U.Fragment = o, U.jsx = v, U.jsxs = v, U;
}
var L = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var We;
function Tr() {
  return We || (We = 1, process.env.NODE_ENV !== "production" && function() {
    var r = Ne, n = Symbol.for("react.element"), o = Symbol.for("react.portal"), i = Symbol.for("react.fragment"), u = Symbol.for("react.strict_mode"), f = Symbol.for("react.profiler"), v = Symbol.for("react.provider"), b = Symbol.for("react.context"), g = Symbol.for("react.forward_ref"), R = Symbol.for("react.suspense"), y = Symbol.for("react.suspense_list"), O = Symbol.for("react.memo"), h = Symbol.for("react.lazy"), k = Symbol.for("react.offscreen"), I = Symbol.iterator, P = "@@iterator";
    function F(e) {
      if (e === null || typeof e != "object")
        return null;
      var t = I && e[I] || e[P];
      return typeof t == "function" ? t : null;
    }
    var W = r.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function j(e) {
      {
        for (var t = arguments.length, a = new Array(t > 1 ? t - 1 : 0), s = 1; s < t; s++)
          a[s - 1] = arguments[s];
        Ve("error", e, a);
      }
    }
    function Ve(e, t, a) {
      {
        var s = W.ReactDebugCurrentFrame, d = s.getStackAddendum();
        d !== "" && (t += "%s", a = a.concat([d]));
        var p = a.map(function(l) {
          return String(l);
        });
        p.unshift("Warning: " + t), Function.prototype.apply.call(console[e], console, p);
      }
    }
    var Be = !1, qe = !1, Je = !1, Ke = !1, Ge = !1, fe;
    fe = Symbol.for("react.module.reference");
    function He(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === i || e === f || Ge || e === u || e === R || e === y || Ke || e === k || Be || qe || Je || typeof e == "object" && e !== null && (e.$$typeof === h || e.$$typeof === O || e.$$typeof === v || e.$$typeof === b || e.$$typeof === g || // This needs to include all possible module reference object
      // types supported by any Flight configuration anywhere since
      // we don't know which Flight build this will end up being used
      // with.
      e.$$typeof === fe || e.getModuleId !== void 0));
    }
    function ze(e, t, a) {
      var s = e.displayName;
      if (s)
        return s;
      var d = t.displayName || t.name || "";
      return d !== "" ? a + "(" + d + ")" : a;
    }
    function de(e) {
      return e.displayName || "Context";
    }
    function S(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && j("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case i:
          return "Fragment";
        case o:
          return "Portal";
        case f:
          return "Profiler";
        case u:
          return "StrictMode";
        case R:
          return "Suspense";
        case y:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case b:
            var t = e;
            return de(t) + ".Consumer";
          case v:
            var a = e;
            return de(a._context) + ".Provider";
          case g:
            return ze(e, e.render, "ForwardRef");
          case O:
            var s = e.displayName || null;
            return s !== null ? s : S(e.type) || "Memo";
          case h: {
            var d = e, p = d._payload, l = d._init;
            try {
              return S(l(p));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var A = Object.assign, M = 0, ve, pe, ge, he, ye, me, Ee;
    function be() {
    }
    be.__reactDisabledLog = !0;
    function Xe() {
      {
        if (M === 0) {
          ve = console.log, pe = console.info, ge = console.warn, he = console.error, ye = console.group, me = console.groupCollapsed, Ee = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: be,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        M++;
      }
    }
    function Ze() {
      {
        if (M--, M === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: A({}, e, {
              value: ve
            }),
            info: A({}, e, {
              value: pe
            }),
            warn: A({}, e, {
              value: ge
            }),
            error: A({}, e, {
              value: he
            }),
            group: A({}, e, {
              value: ye
            }),
            groupCollapsed: A({}, e, {
              value: me
            }),
            groupEnd: A({}, e, {
              value: Ee
            })
          });
        }
        M < 0 && j("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var z = W.ReactCurrentDispatcher, X;
    function B(e, t, a) {
      {
        if (X === void 0)
          try {
            throw Error();
          } catch (d) {
            var s = d.stack.trim().match(/\n( *(at )?)/);
            X = s && s[1] || "";
          }
        return `
` + X + e;
      }
    }
    var Z = !1, q;
    {
      var Qe = typeof WeakMap == "function" ? WeakMap : Map;
      q = new Qe();
    }
    function Re(e, t) {
      if (!e || Z)
        return "";
      {
        var a = q.get(e);
        if (a !== void 0)
          return a;
      }
      var s;
      Z = !0;
      var d = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var p;
      p = z.current, z.current = null, Xe();
      try {
        if (t) {
          var l = function() {
            throw Error();
          };
          if (Object.defineProperty(l.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(l, []);
            } catch (w) {
              s = w;
            }
            Reflect.construct(e, [], l);
          } else {
            try {
              l.call();
            } catch (w) {
              s = w;
            }
            e.call(l.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (w) {
            s = w;
          }
          e();
        }
      } catch (w) {
        if (w && s && typeof w.stack == "string") {
          for (var c = w.stack.split(`
`), x = s.stack.split(`
`), m = c.length - 1, E = x.length - 1; m >= 1 && E >= 0 && c[m] !== x[E]; )
            E--;
          for (; m >= 1 && E >= 0; m--, E--)
            if (c[m] !== x[E]) {
              if (m !== 1 || E !== 1)
                do
                  if (m--, E--, E < 0 || c[m] !== x[E]) {
                    var C = `
` + c[m].replace(" at new ", " at ");
                    return e.displayName && C.includes("<anonymous>") && (C = C.replace("<anonymous>", e.displayName)), typeof e == "function" && q.set(e, C), C;
                  }
                while (m >= 1 && E >= 0);
              break;
            }
        }
      } finally {
        Z = !1, z.current = p, Ze(), Error.prepareStackTrace = d;
      }
      var $ = e ? e.displayName || e.name : "", De = $ ? B($) : "";
      return typeof e == "function" && q.set(e, De), De;
    }
    function er(e, t, a) {
      return Re(e, !1);
    }
    function rr(e) {
      var t = e.prototype;
      return !!(t && t.isReactComponent);
    }
    function J(e, t, a) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return Re(e, rr(e));
      if (typeof e == "string")
        return B(e);
      switch (e) {
        case R:
          return B("Suspense");
        case y:
          return B("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case g:
            return er(e.render);
          case O:
            return J(e.type, t, a);
          case h: {
            var s = e, d = s._payload, p = s._init;
            try {
              return J(p(d), t, a);
            } catch {
            }
          }
        }
      return "";
    }
    var K = Object.prototype.hasOwnProperty, _e = {}, Oe = W.ReactDebugCurrentFrame;
    function G(e) {
      if (e) {
        var t = e._owner, a = J(e.type, e._source, t ? t.type : null);
        Oe.setExtraStackFrame(a);
      } else
        Oe.setExtraStackFrame(null);
    }
    function tr(e, t, a, s, d) {
      {
        var p = Function.call.bind(K);
        for (var l in e)
          if (p(e, l)) {
            var c = void 0;
            try {
              if (typeof e[l] != "function") {
                var x = Error((s || "React class") + ": " + a + " type `" + l + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[l] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw x.name = "Invariant Violation", x;
              }
              c = e[l](t, l, s, a, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (m) {
              c = m;
            }
            c && !(c instanceof Error) && (G(d), j("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", s || "React class", a, l, typeof c), G(null)), c instanceof Error && !(c.message in _e) && (_e[c.message] = !0, G(d), j("Failed %s type: %s", a, c.message), G(null));
          }
      }
    }
    var nr = Array.isArray;
    function Q(e) {
      return nr(e);
    }
    function ar(e) {
      {
        var t = typeof Symbol == "function" && Symbol.toStringTag, a = t && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return a;
      }
    }
    function or(e) {
      try {
        return je(e), !1;
      } catch {
        return !0;
      }
    }
    function je(e) {
      return "" + e;
    }
    function xe(e) {
      if (or(e))
        return j("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", ar(e)), je(e);
    }
    var Y = W.ReactCurrentOwner, ir = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, Ce, Te, ee;
    ee = {};
    function sr(e) {
      if (K.call(e, "ref")) {
        var t = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (t && t.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function ur(e) {
      if (K.call(e, "key")) {
        var t = Object.getOwnPropertyDescriptor(e, "key").get;
        if (t && t.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function cr(e, t) {
      if (typeof e.ref == "string" && Y.current && t && Y.current.stateNode !== t) {
        var a = S(Y.current.type);
        ee[a] || (j('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', S(Y.current.type), e.ref), ee[a] = !0);
      }
    }
    function lr(e, t) {
      {
        var a = function() {
          Ce || (Ce = !0, j("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", t));
        };
        a.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: a,
          configurable: !0
        });
      }
    }
    function fr(e, t) {
      {
        var a = function() {
          Te || (Te = !0, j("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", t));
        };
        a.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: a,
          configurable: !0
        });
      }
    }
    var dr = function(e, t, a, s, d, p, l) {
      var c = {
        // This tag allows us to uniquely identify this as a React Element
        $$typeof: n,
        // Built-in properties that belong on the element
        type: e,
        key: t,
        ref: a,
        props: l,
        // Record the component responsible for creating this element.
        _owner: p
      };
      return c._store = {}, Object.defineProperty(c._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(c, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: s
      }), Object.defineProperty(c, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: d
      }), Object.freeze && (Object.freeze(c.props), Object.freeze(c)), c;
    };
    function vr(e, t, a, s, d) {
      {
        var p, l = {}, c = null, x = null;
        a !== void 0 && (xe(a), c = "" + a), ur(t) && (xe(t.key), c = "" + t.key), sr(t) && (x = t.ref, cr(t, d));
        for (p in t)
          K.call(t, p) && !ir.hasOwnProperty(p) && (l[p] = t[p]);
        if (e && e.defaultProps) {
          var m = e.defaultProps;
          for (p in m)
            l[p] === void 0 && (l[p] = m[p]);
        }
        if (c || x) {
          var E = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          c && lr(l, E), x && fr(l, E);
        }
        return dr(e, c, x, d, s, Y.current, l);
      }
    }
    var re = W.ReactCurrentOwner, Pe = W.ReactDebugCurrentFrame;
    function N(e) {
      if (e) {
        var t = e._owner, a = J(e.type, e._source, t ? t.type : null);
        Pe.setExtraStackFrame(a);
      } else
        Pe.setExtraStackFrame(null);
    }
    var te;
    te = !1;
    function ne(e) {
      return typeof e == "object" && e !== null && e.$$typeof === n;
    }
    function Se() {
      {
        if (re.current) {
          var e = S(re.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function pr(e) {
      {
        if (e !== void 0) {
          var t = e.fileName.replace(/^.*[\\\/]/, ""), a = e.lineNumber;
          return `

Check your code at ` + t + ":" + a + ".";
        }
        return "";
      }
    }
    var we = {};
    function gr(e) {
      {
        var t = Se();
        if (!t) {
          var a = typeof e == "string" ? e : e.displayName || e.name;
          a && (t = `

Check the top-level render call using <` + a + ">.");
        }
        return t;
      }
    }
    function ke(e, t) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var a = gr(t);
        if (we[a])
          return;
        we[a] = !0;
        var s = "";
        e && e._owner && e._owner !== re.current && (s = " It was passed a child from " + S(e._owner.type) + "."), N(e), j('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', a, s), N(null);
      }
    }
    function Fe(e, t) {
      {
        if (typeof e != "object")
          return;
        if (Q(e))
          for (var a = 0; a < e.length; a++) {
            var s = e[a];
            ne(s) && ke(s, t);
          }
        else if (ne(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var d = F(e);
          if (typeof d == "function" && d !== e.entries)
            for (var p = d.call(e), l; !(l = p.next()).done; )
              ne(l.value) && ke(l.value, t);
        }
      }
    }
    function hr(e) {
      {
        var t = e.type;
        if (t == null || typeof t == "string")
          return;
        var a;
        if (typeof t == "function")
          a = t.propTypes;
        else if (typeof t == "object" && (t.$$typeof === g || // Note: Memo only checks outer props here.
        // Inner props are checked in the reconciler.
        t.$$typeof === O))
          a = t.propTypes;
        else
          return;
        if (a) {
          var s = S(t);
          tr(a, e.props, "prop", s, e);
        } else if (t.PropTypes !== void 0 && !te) {
          te = !0;
          var d = S(t);
          j("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", d || "Unknown");
        }
        typeof t.getDefaultProps == "function" && !t.getDefaultProps.isReactClassApproved && j("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function yr(e) {
      {
        for (var t = Object.keys(e.props), a = 0; a < t.length; a++) {
          var s = t[a];
          if (s !== "children" && s !== "key") {
            N(e), j("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", s), N(null);
            break;
          }
        }
        e.ref !== null && (N(e), j("Invalid attribute `ref` supplied to `React.Fragment`."), N(null));
      }
    }
    function Ae(e, t, a, s, d, p) {
      {
        var l = He(e);
        if (!l) {
          var c = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (c += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var x = pr(d);
          x ? c += x : c += Se();
          var m;
          e === null ? m = "null" : Q(e) ? m = "array" : e !== void 0 && e.$$typeof === n ? (m = "<" + (S(e.type) || "Unknown") + " />", c = " Did you accidentally export a JSX literal instead of a component?") : m = typeof e, j("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", m, c);
        }
        var E = vr(e, t, a, d, p);
        if (E == null)
          return E;
        if (l) {
          var C = t.children;
          if (C !== void 0)
            if (s)
              if (Q(C)) {
                for (var $ = 0; $ < C.length; $++)
                  Fe(C[$], e);
                Object.freeze && Object.freeze(C);
              } else
                j("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              Fe(C, e);
        }
        return e === i ? yr(E) : hr(E), E;
      }
    }
    function mr(e, t, a) {
      return Ae(e, t, a, !0);
    }
    function Er(e, t, a) {
      return Ae(e, t, a, !1);
    }
    var br = Er, Rr = mr;
    L.Fragment = i, L.jsx = br, L.jsxs = Rr;
  }()), L;
}
process.env.NODE_ENV === "production" ? oe.exports = Cr() : oe.exports = Tr();
var _ = oe.exports;
const Me = _r(
  void 0
);
function H() {
  const r = Or(Me);
  return r != null && r.isAlive ? r : void 0;
}
function Wr() {
  const r = H(), [n, o] = V();
  return D(() => {
    if (!r || !r.isAlive) {
      o(void 0);
      return;
    }
    const i = r.unreads.onChange((u) => {
      o(u);
    });
    return () => i.unsubscribe();
  }, [r]), n;
}
function ie({ target: r, handlers: n }) {
  return r != null && r.isAlive ? Object.keys(n).filter((i) => i.startsWith("on")).filter((i) => typeof n[i] == "function").map((i) => /* @__PURE__ */ _.jsx(
    Pr,
    {
      name: i,
      handler: n[i],
      target: r
    },
    i
  )) : null;
}
function Pr(r) {
  const { handler: n } = r;
  return D(() => {
    var f, v;
    if (!((f = r.target) != null && f.isAlive))
      return;
    const { name: o, target: i } = r.name === "onUnreadsChange" ? { name: "onChange", target: r.target.unreads } : { name: r.name, target: r.target };
    if (typeof i[o] != "function") {
      console.warn(
        `[@talkjs/react] Trying to handle event ${o} which does not exist on ${(v = i == null ? void 0 : i.constructor) == null ? void 0 : v.name}. Did you make a typo?`
      );
      return;
    }
    const u = i[o](n);
    return () => {
      u.unsubscribe();
    };
  }, [r.target, r.name, n]), null;
}
var Sr = function r(n, o) {
  if (n === o)
    return !0;
  if (n && o && typeof n == "object" && typeof o == "object") {
    if (n.constructor !== o.constructor)
      return !1;
    var i, u, f;
    if (Array.isArray(n)) {
      if (i = n.length, i != o.length)
        return !1;
      for (u = i; u-- !== 0; )
        if (!r(n[u], o[u]))
          return !1;
      return !0;
    }
    if (n.constructor === RegExp)
      return n.source === o.source && n.flags === o.flags;
    if (n.valueOf !== Object.prototype.valueOf)
      return n.valueOf() === o.valueOf();
    if (n.toString !== Object.prototype.toString)
      return n.toString() === o.toString();
    if (f = Object.keys(n), i = f.length, i !== Object.keys(o).length)
      return !1;
    for (u = i; u-- !== 0; )
      if (!Object.prototype.hasOwnProperty.call(o, f[u]))
        return !1;
    for (u = i; u-- !== 0; ) {
      var v = f[u];
      if (!r(n[v], o[v]))
        return !1;
    }
    return !0;
  }
  return n !== n && o !== o;
};
const wr = /* @__PURE__ */ xr(Sr);
function Ye(r) {
  const n = $e();
  return (!n.current || !wr(r, n.current.val)) && (n.current = { val: r }), n.current.val;
}
function T(r, n, o) {
  n = Ye(n), D(() => {
    n !== void 0 && (r != null && r.isAlive) && r[o](n);
  }, [o, r, n]);
}
function se(r, n, o, i) {
  const u = jr(() => typeof o == "function" ? r != null && r.isAlive ? o(r) : void 0 : o ?? i, [r, o, i]);
  T(n, u, "select");
}
function ue(r, n, o, i) {
  const [u, f] = V();
  return o = Ye(o), D(() => {
    if (r != null && r.isAlive) {
      const v = r[n](o);
      return f(v), i && (i.current = v), () => {
        v.destroy(), f(void 0);
      };
    } else
      f(void 0), i && (i.current = void 0);
  }, [r, n, o]), u;
}
function Ue(r, n) {
  const [o, i] = V(!1);
  return D(() => {
    r != null && r.isAlive && r.mount(n).then(() => i(!0));
  }, [r]), o;
}
function Nr(r) {
  const [n, o] = V(!1), [i, u] = V(), { userId: f, syncUser: v, appId: b, signature: g, sessionRef: R, desktopNotificationEnabled: y } = r;
  return D(() => {
    ae.ready.then(() => o(!0));
  }, []), D(() => {
    if (n) {
      const O = typeof v == "function" ? v() : v ?? new ae.User(f), h = new ae.Session({ appId: b, me: O, signature: g });
      return u(h), R && (R.current = h), () => {
        h.destroy(), u(void 0), R && (R.current = void 0);
      };
    }
  }, [n, b, f, v, R]), T(
    i,
    y,
    "setDesktopNotificationEnabled"
  ), /* @__PURE__ */ _.jsxs(_.Fragment, { children: [
    /* @__PURE__ */ _.jsx(Me.Provider, { value: i, children: r.children }),
    /* @__PURE__ */ _.jsx(ie, { target: i, handlers: r })
  ] });
}
const ce = (() => {
  const r = /* @__PURE__ */ new WeakMap();
  let n = 0;
  return function(i) {
    return r.has(i) || r.set(i, ++n), r.get(i);
  };
})();
function le(r, n) {
  const o = {}, i = {};
  for (const [u, f] of Object.entries(r))
    u.startsWith(n) ? o[u] = f : i[u] = f;
  return [o, i];
}
function Le(r) {
  const { box: n, loadingComponent: o, className: i, handlers: u } = r, f = $e(null), v = Ue(n, f.current), b = v ? r.style : { ...r.style, display: "none" };
  return /* @__PURE__ */ _.jsxs(_.Fragment, { children: [
    !v && /* @__PURE__ */ _.jsx("div", { style: r.style, className: i, children: o }),
    /* @__PURE__ */ _.jsx("div", { ref: f, style: b, className: i }),
    /* @__PURE__ */ _.jsx(ie, { target: n, handlers: u })
  ] });
}
function $r(r) {
  const n = H();
  if (n) {
    const o = ce(n);
    return /* @__PURE__ */ _.jsx(kr, { session: n, ...r }, o);
  } else
    return /* @__PURE__ */ _.jsx("div", { style: r.style, className: r.className, children: r.loadingComponent });
}
function kr(r) {
  const {
    session: n,
    conversationId: o,
    syncConversation: i,
    chatboxRef: u,
    style: f,
    className: v,
    loadingComponent: b,
    ...g
  } = r, [R, y] = le(g, "on"), { messageFilter: O, presence: h, highlightedWords: k, ...I } = y, P = ue(n, "createChatbox", I, u);
  return T(P, O, "setMessageFilter"), T(P, h, "setPresence"), T(P, k, "setHighlightedWords"), se(n, P, i, o), /* @__PURE__ */ _.jsx(
    Le,
    {
      box: P,
      session: n,
      className: v,
      style: f,
      loadingComponent: b,
      handlers: R
    }
  );
}
function Mr(r) {
  const n = H();
  if (n) {
    const o = ce(n);
    return /* @__PURE__ */ _.jsx(Fr, { session: n, ...r }, o);
  } else
    return /* @__PURE__ */ _.jsx("div", { style: r.style, className: r.className, children: r.loadingComponent });
}
function Fr(r) {
  const {
    session: n,
    conversationId: o,
    syncConversation: i,
    inboxRef: u,
    style: f,
    className: v,
    loadingComponent: b,
    ...g
  } = r, [R, y] = le(g, "on"), {
    messageFilter: O,
    feedFilter: h,
    presence: k,
    highlightedWords: I,
    ...P
  } = y, F = ue(n, "createInbox", P, u);
  return T(F, O, "setMessageFilter"), T(F, h, "setFeedFilter"), T(F, k, "setPresence"), T(F, I, "setHighlightedWords"), se(n, F, i, o), /* @__PURE__ */ _.jsx(
    Le,
    {
      box: F,
      session: n,
      className: v,
      style: f,
      loadingComponent: b,
      handlers: R
    }
  );
}
function Yr(r) {
  const n = H();
  if (n) {
    const o = ce(n);
    return /* @__PURE__ */ _.jsx(Ar, { session: n, ...r }, o);
  }
  return null;
}
function Ar(r) {
  const {
    session: n,
    conversationId: o,
    syncConversation: i,
    popupRef: u,
    ...f
  } = r, [v, b] = le(f, "on"), { messageFilter: g, presence: R, highlightedWords: y, ...O } = b, h = ue(n, "createPopup", O, u);
  return T(h, g, "setMessageFilter"), T(h, R, "setPresence"), T(h, y, "setHighlightedWords"), se(n, h, i, o), Ue(h, void 0), /* @__PURE__ */ _.jsx(ie, { target: h, handlers: v });
}
export {
  $r as Chatbox,
  Mr as Inbox,
  Yr as Popup,
  Nr as Session,
  H as useSession,
  Wr as useUnreads
};
